packages/runtime/runtime-definitions/src/dataStoreContext.ts

```diff
@@ -346,14 +346,52 @@ export interface IFluidDataStoreContextEvents extends IEvent {
 	(event: "attaching" | "attached", listener: () => void);
 }
 
+export interface IProvideExperimentalFluidGCInfo {
```

Comment:
I am wondering if this should be moved to`container-runtime-definitions` or `container-runtime` since its confined to that layer. `runtime-definitions` are for things that are used within the whole runtime layer including data store runtime.

Aqueduct installs both `container-runtime-definitions` and `container-runtime` so its equivalent from that perspective.

Reply:
haven't look at all of this, but i would reconsider using the provider pattern here at all. Its really not meant for exposing internals like this, and the few usages i see would be better as just an optional property.

Reply:
i could really use some scenario context here, as the description is all about implementation.

Reply:
An app will do handle.get and need some way to inspect the result to find this stuff. That's why we went with this approach. Open to other suggestions.

Reply:
i don't think app should inspect the things they get from the handle for data like this. what they get from the handle is the applications object, which also requires realization. this feels much more like system data.

Reply:
I also curious what the app would do after inspection, how is the data used?

Reply:
Telemetry to understand when GC-impacting bugs are present

Reply:
I would also consider just exposing this at the context/runtime, and letting the app choose how to get it out if they need it, rather than making it a top level concept for all consumers

Reply:
also, don't we already provide that telemetry? why do they need it again?

Reply:
It's an example of providing an Observability API so they don't take a hard dependency on our logs. It lets them instrument in a more consistent manner at the scenario level, our logs are not quite tuned to what they care about. 

And I think one specific issue is the callstack we generate doesn't reach high enough, so they're going to try generating the stack from their code when they get back an inactive/Tombstoned thing.